test/pair.test.ts

```diff
@@ -1,12 +1,85 @@
-import { Token, Pair } from '../src/entities'
+import { Token, Pair, TokenAmount, WETH } from '../src/entities'
 import { ChainId } from '../src/constants'
 
 describe('Pair', () => {
+  const USDC = new Token(ChainId.MAINNET, '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48', 18, 'USDC', 'USD Coin')
+  const DAI = new Token(ChainId.MAINNET, '0x6B175474E89094C44Da98b954EedeAC495271d0F', 18, 'DAI', 'DAI Stablecoin')
+
+  describe('constructor', () => {
+    it('cannot be used for tokens on different chains', () => {
+      expect(() => new Pair(new TokenAmount(USDC, '100'), new TokenAmount(WETH[ChainId.RINKEBY], '100'))).toThrow(
+        'CHAIN_IDS'
+      )
+    })
+  })
+
   describe('#getAddress', () => {
     it('returns the correct address', () => {
-      const usdc = new Token(ChainId.MAINNET, '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48', 18, 'USDC', 'USD Coin')
-      const dai = new Token(ChainId.MAINNET, '0x6B175474E89094C44Da98b954EedeAC495271d0F', 18, 'DAI', 'DAI Stablecoin')
-      expect(Pair.getAddress(usdc, dai)).toEqual('0xAE461cA67B15dc8dc81CE7615e0320dA1A9aB8D5')
+      expect(Pair.getAddress(USDC, DAI)).toEqual('0xAE461cA67B15dc8dc81CE7615e0320dA1A9aB8D5')
```

Comment:
oh i was thinking, this should probably return a token. maybe keep the method with intent to deprecate and add a `Pair.getLiquidityToken` or something?

Reply:
maybe not worth though idk

Reply:
not sure, maybe consider in another pr. the way we use it in the interface in both cases, creating token instances would be a waste